src/modules/multipart/multipartx.js

```diff
@@ -32,9 +32,18 @@ class MultipartX {
   fromString (jsonString) {
     let chunks = []
     // ToDo:: FLODATA_MAX_LEN vs CHOP_MAX_LEN
-    while (jsonString.length > CHOP_MAX_LEN) {
-      chunks.push(jsonString.slice(0, CHOP_MAX_LEN))
-      jsonString = jsonString.slice(CHOP_MAX_LEN)
+    let maximumLength = CHOP_MAX_LEN
+    while (jsonString.length > maximumLength) {
+      // If we have between 100 and 999 parts, our max length needs to be reduced by one
+      if (chunks.length >= 99) { maximumLength = CHOP_MAX_LEN - 1 }
+      // If we have between 1000 and 9999 parts, our max length needs to be reduced by one
+      if (chunks.length >= 999) { maximumLength = CHOP_MAX_LEN - 2 }
+      // If we have between 10000 and 99999 parts, our max length needs to be reduced by one
+      if (chunks.length >= 9999) { maximumLength = CHOP_MAX_LEN - 3 }
```

Comment:
Impressive multipart. Has the impact upon oipd been tested?

Reply:
I have not yet tested out the impact on OIPd for such a large mulitpart. The largest that I witnessed for our testing record set was a multipart of ~350 in length.